pkg/kubelet/pleg/generic.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pleg
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// GenericPLEG is an extremely simple generic PLEG that relies solely on
+// periodic listing to discover container changes. It should be be used
+// as temporary replacement or container runtimes do not support a proper
+// event generator yet.
+//
+// Note that Generic PLEG assumes that a container would not be created and
+// garbage collected within one relist period. If this is not true, it might
+// miss the container completely. Relisting failure would also lead to a longer
+// window where a container could be missing. It is recommended to set the
+// relist period short and have an auxiliary, longer periodic sync in kubelet
+// as the safety net.
+type GenericPLEG struct {
+	// The period for relisting.
+	relistPeriod time.Duration
+	runtime      kubecontainer.Runtime
+	eventChannel chan *PodLifecycleEvent
+	// IDs of all visible containers.
+	containers sets.String
+	// IDs of all running containers
+	runningContainers sets.String
+	// Map container ID to pod ID.
+	idMap map[string]types.UID
+}
+
+var _ PodLifecycleEventGenerator = &GenericPLEG{}
+
+func NewGenericPLEG(runtime kubecontainer.Runtime, channelCapacity int,
+	relistPeriod time.Duration) *GenericPLEG {
```

Comment:
I assume you are referring to unit tests of this object? If thats true, why can't unit tests not use this factory method?